### DISCUSSION TITLE
20230627 회의

### DISCUSSION THREAD

**Discussion by LeeHyungJoo** [KKangTaee/HifeSurvival#11] [General]
**##GameMode** 
클라이언트에서 Spawn 다 완료된 시점 알아야함.
즉, 게임 플레이할 준비를 다 완료 했다. 
Spawn 이 실패한 클라이언트가 Id 값에 기반한 갱신 패킷 내리면 오류

**##Collision**
Body Range 있어야 함. 
스킬 관련 작업 시 필요할 것 같음.  

**##동일 좌표 계산 하는 것이 필요.**
Player는 클라 기준임 완전. 
Monster 는 서버 기준. 

**##좌표 동등 연산 기준** 
좀 더 정확하게 수학적으로 계산 필요. 
현재는 그냥 거리가 0.25 이하면 같다고 판단하고 있음. 
클라틱, 서버틱, monster AI 틱 모두 충족시킬 수 있는 처리가 필요. 
아니면 앞선 틱들을 어느정도 정렬하는 방안도 있음. 

**##동기화**
클라에서는 Time line 을 만들어 대소 비교만 
그리고 요청하고 응답 받을 시에만 서버 시간을 내리면 어떨까? 
실제로 클라이언트에서 시간이 가는 것과, 서버에서 내리는 그 시점 값 (timestamp)
을 비교해서, 지금 latency가 어느정도 있는지 확인할 수 있고 이를 통해 보정할 수 있을 것 같음. 
어느정도 오차는 허용한 후 오차 벗어날 경우. 계속 보정. (서버시간과 동기화)

**##플레이어 공격 가능 체크** 
서버에서도 해야함. 

**##죽은 후 처리** 
리스폰 시간 동안 죽기전에 입력한 방향으로 계속 움직임
서버와 통신은 되지 않는 듯 함. 

**##targetid ->id**
스킬에 따라서 헷갈리지 않게 네이밍 바꿀 필요가 있음
targetid -> 내가 지금 타겟으로 한 녀석 
id -> 내 자신 id 

**##enum 제일 끝에 none이 있는 이유?**

**Comment by KKangTaee** [comment]
2023.06.27 회의
- 

## 해야할 일

### 아이템 드랍 & 드롭 (서버, 클라)
 * 몬스터 사망 시 일정한 확률을 바탕으로 아이템을 드랍하게 되어있음.
 * 아이템 드랍에 대한 정보는 monsters 시트의 'rewards' 를 참조함.
 * rewardIds는 다음과 같은 정보를 가지고 있음.
   * 타입 : 갯수 : 확률 
   * 타입은 골드, 아이탬 을 정의함.
   * 확률 같은 경우에는 100을 기준으로 확률을 정의 함. 만약 40일 경우에는 40% 의 확률로 드랍.
 * 현재 아이탬 드랍 시 S_DropItem을 내려주게 되어있음.
 * 그리고, 플레이어가 아이템을 획득하면 C_PickItem 을 서버로 올림
 * 그 후, 플레이어의 획득했음을 S_GetItem으로 내려줌.
  * 여기서 고려해야할 부분은 획득한 당사자와, 그렇지 않은 사람들을 어떤식으로 처리해야할지 생각이 필요
  * 맵에 아이템이 획득되어 모든사람들이 맵에 놓인 아이템 오브젝트를 없애기 위해서는 전체적으로 내릴 필요가 있음.
  * 다른 플레이어가 장착한 아이템을 월드UI상에 노출시키는 것이 기획의 의도이기 때문에, 그렇게 나누지 말고 다 내려주면 좋지 않을까..? 라는 생각.
 * 위의 과정들을 서버의 기호에 맞게 수정할 필요가 있음.
### 스텟 레벨업 (서버, 클라)
  * 현재는 CS_UpdateStat 이라는 코드를 통해서 처리하도록 만들어놓음
  *  내부의 멤버가 stat 하나로 구현되어있는데, 이걸 현재스탯, 추가 스탯으로 분리할 필요가 있음.
  * 해당 패킷의 명칭도 모호하다고 생각되기 때문에 이름도 변경될 필요가 있음
### 죽은 후 사망처리 버그 (클라)
  * 사망 시 플레이어의 디패드가 정지상태가 아닌 경우가 발생. 디패드 초기화 부분을 추가로 넣어야함.
### TargetId 를 id로 수정하고, 번호로 나누기 (서버)
    1. 플레이어 1-100
    2. 몬스터 1000 이상
### 게임 시작 플로우 처리 (서버, 클라)
   * 현재 모두 인게임 진입 후, 모든 유저가 인게임에 진입을 했는지 체크할 방법이 없음.
   * 이럴경우 디바이스의 성능이 안좋거나 통신처리가 안좋아 늦게 진입한 유저는 불리함.
   * 모든 유저가 다같이 진입했을 때 게임을 시작하겠음 클라에서 인게임 진입 완료를 전송하는 패킷이 필요
### 월드맵 수정 (클라)
### BodyRange 스태틱 시트 몬스터, 캐릭터 추가 (클라)
   * 논타겟 판정을 위한 플레이어, 몬스터의 바디범위가 필요함.
   * Q. BodyRange의 중점 설정에 대해 좀 생각을 해봐야 함. (일단, pos로 하면 안될듯)
### 길찾기 라이브러리 구현 (클라)
### 전장의 안개 (클라)

**Reply by LeeHyungJoo** [reply]
대역대 기준 2개로만 유지 . #13

**Reply by LeeHyungJoo** [reply]
아이템 드랍 & 드롭 (서버, 클라) #16  이슈 발급

**Reply by LeeHyungJoo** [reply]
스텟 레벨업 (서버, 클라) #15 이슈 발급
